game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProUtils.java

```diff
@@ -270,4 +273,14 @@ public static boolean isPassiveNeutralPlayer(final GamePlayer player) {
         .filter(s -> player.equals(s.getPlayerId()))
         .noneMatch(s -> GameStep.isCombatMoveStep(s.getName()));
   }
+
+  public static String summarizeUnits(Collection<Unit> units) {
+    IntegerMap<String> counts = new IntegerMap<>();
+    for (Unit u : units) {
+      counts.add(u.toString(), 1);
+    }
+    return counts.entrySet().stream()
```

Comment:
A comment here might be really helpful to give an example of the output string.
This method is a good candidate for a unit test.

Reply:
Good idea on both fronts, done.